src/blockchain_db/lmdb/db_lmdb.cpp

```diff
@@ -472,6 +472,32 @@ void mdb_txn_safe::increment_txns(int i)
 	num_active_txns += i;
 }
 
+#define TXN_PREFIX(flags); \
+  mdb_txn_safe auto_txn; \
+  mdb_txn_safe* txn_ptr = &auto_txn; \
+  if (m_batch_active) \
+    txn_ptr = m_write_txn; \
+  else \
+  { \
+    if (auto mdb_res = lmdb_txn_begin(m_env, NULL, flags, auto_txn)) \
+      throw0(DB_ERROR(lmdb_error(std::string("Failed to create a transaction for the db in ")+__FUNCTION__+": ", mdb_res).c_str())); \
+  } \
+
+#define TXN_PREFIX_RDONLY() \
+  MDB_txn *m_txn; \
+  mdb_txn_cursors *m_cursors; \
+  mdb_txn_safe auto_txn; \
```

Comment:
Is this still needed if `block_rtxn_start` uses it too now? This should be why the code worked in most situations - the bulk of the calls were using `TXN_PREFIX_RDONLY`.

I don't think it hurts to double up, but it looks unnecessary.

Reply:
Oooops, looks like you commented on this above already. I figured it would work, but I think you'd have to go back through and do a major re-haul to "get it correct", whereas this needs push sooner rather than later.